lib/cdo/delete_accounts_helper.rb

```diff
@@ -0,0 +1,184 @@
+require_relative '../../shared/middleware/helpers/storage_id'
+require 'cdo/aws/s3'
+require 'cdo/solr'
+require 'cdo/solr_helper'
+
+module DeleteAccountsHelper
+  ANONYMIZED_SECTION_NAME = 'system_deleted'.freeze
+  OPEN_ENDED_LEVEL_TYPES = %w(
+    Applab
+    FreeResponse
+    Gamelab
+    Weblab
+  ).freeze
+
+  raise 'No SOLR server configured' unless CDO.solr_server
+  SOLR = Solr::Server.new(host: CDO.solr_server)
+
+  # Deletes all project-backed progress associated with a user.
+  # @param [Integer] user_id The user to delete the project-backed progress of.
+  def self.delete_project_backed_progress(user_id)
+    # Query the DB for the user's storage ID.
+    user_storage_ids_row = PEGASUS_DB[:user_storage_ids].where(user_id: user_id).first
+    return unless user_storage_ids_row
+    storage_id = user_storage_ids_row[:id]
+
+    # Delete project data stored on AWS s3.
+    s3_client = AWS::S3.create_client
+    {
+      CDO.animations_s3_bucket => CDO.animations_s3_directory,
+      CDO.assets_s3_bucket => CDO.assets_s3_directory,
+      CDO.files_s3_bucket => CDO.files_s3_directory,
+      CDO.sources_s3_bucket => CDO.sources_s3_directory
+    }.each do |bucket, directory|
+      unless bucket.present? && directory.present?
+        raise "Missing AWS s3 bucket information (bucket: #{bucket}, directory: #{directory})."
+      end
+      s3_client.buckets[bucket].objects.with_prefix("/#{directory}/#{storage_id}/").delete_all
+    end
+
+    # Query the DB for the channel IDs associated with the storage ID, deleting data on Firebase
+    # for each.
+    PEGASUS_DB[:storage_apps].where(storage_id: storage_id).each do |storage_app|
+      encrypted_channel_id = storage_encrypt_channel_id storage_id, storage_app[:id]
+      # TODO(asher): This makes more sense as
+      #   FirebaseHelper.new.delete_channel(encrypted_channel_id).
+      # Refactor FirebaseHelper to allow this.
+      FirebaseHelper.new(encrypted_channel_id).delete_channel
+    end
+  end
+
+  # Cleans all LevelSource-backed progress associated with the user.
+  # WARNING: Since the relationship between user_levels and activities and level_sources is
+  #   many-to-one, we cannot be certain that this is not messing with the integrity of the data for
+  #   other users.
```

Comment:
Per offline discussion with @jeremydstone, this has been changed to severe the link between the user's level-backed progress and the user (see commit d39f371) rather than remove the actual student work (image). PTAL.

FYI @poorvasingal @Hamms @jeremydstone.